rpm/files.go

```diff
@@ -0,0 +1,102 @@
+package rpm
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"sync"
+
+	"github.com/quay/claircore"
+	"github.com/quay/zlog"
+)
+
+// filesCache is used for concurrent access to the map containing layer.Hash -> map RPM files.
+// The value is a map to allow for quick member checking.
+type filesCache struct {
+	c  map[string]map[string]struct{}
```

Comment:
It's tempting to use a sync.Map here but I think we want the lock to extend over outer and inner map

Reply:
I am wondering if we should take some inspiration from [`singleflight`](https://pkg.go.dev/golang.org/x/sync/singleflight) here. Lock per key (layer) and then cleanup once nobody needs it anymore (perhaps similar to the gc we have now. track all the contexts and then cleanup once all known, related contexts are closed).

Not sure if that's work, but if so, I think it'll allow for more concurrency

Reply:
I have simplified the locking a little (now concurrent RPM DBs reads can happen) and expanded the functionality to track all the calling contexts to inform when deletion happens.